storage/metrics/tsicache.go

```diff
@@ -0,0 +1,73 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sort"
+)
+
+const cacheSubsystem = "tsi_cache"     // sub-system associated with TSI index cache.
+
+// CacheMetrics wrap all collectors of metrics from the TSI index cache.
+type CacheMetrics struct {
```

Comment:
If this is for TSI, we should put `TSI` in the name. There is also an inmem write cache that will need to track metrics will probably be called `CacheMetrics`.